### ISSUE TITLE
Inconsistent data received (Bluetooth classic, RFComm, Serialport)

### ISSUE THREAD

**Issue by mdima** [inthehand/32feet#250]
Hi,
   I am trying to connect to a medica device using Visual Studio + MAUI + InTheHand.Net.Bluetooth. 

I successfully pair the device, then connect using the BluetoothService.SerialPort service, but whatever I receive is:

CALL c4:62:ea:98:ce:08 1103 RFCOMM

where c4:62:ea:98:ce:08 is the address of my device. If I try to connect using BluetoothService.DialupNetworking (1103) the connection do not succeed. What I do is: 

```
stream = client.GetStream();
StreamWriter sw = new StreamWriter(stream, System.Text.Encoding.ASCII); (this will be needed later)
StreamReader sr = new StreamReader(stream, System.Text.Encoding.ASCII);
char[] buffer = new char[10000];
await sr.ReadAsync(buffer, 0, (int)stream.Length);
```

Is something I am doing wrong? 

Thanks,
Michele

**Comment by euquiq** [comment]
Hi Michele!

Please read mi "issue" posted a couple of hours ago ... the code in there "at least" is reading correctly thru SerialPort, maybe It can help you detecting what you may be doing wrong / differently ? I do the stream reading with a bit more of code lines / finesse than your approach, I think ? Hopefully it helps. 

Also, maybe you have any hint for my own issue :)


Regards,
Enrique

**Comment by mdima** [comment]
Thank you Enrique, I tried to use the same pattern you used here: https://github.com/inthehand/32feet/issues/252 to read from the device, but the result doesn't change at all. 

About your issue, I am experiencing the same problem. The ".Connected" property gets false only after a read timeout, even if I turn off the device after I connect it. 

Regards,
Michele

**Comment by ajay-mundi** [comment]
Hello Enrique and Michele,

Enrique I also tried using your approach (was a lot nicer than mine lol), but still can't seem to read anything from the stream. Not really sure what I am doing with this bluetooth stuff to be honest. Having a MAUI cross platform sample in this repo would be really helpful.

**Comment by peterfoot** [comment]
This is very odd. What is the platform this code is running on? If Android have you tried running the same code on Windows (or vice versa)?

**Comment by mdima** [comment]
Hi, yes. I am using Android and Windows (I recreated a simple console application that use the same functions of the Android app), but the result is the same.

**Comment by peterfoot** [comment]
Is there a chance that the device is connected to something else, hence this odd message. Have you tried resetting the device (even removing all pairings and starting from fresh)?

**Comment by mdima** [comment]
mmmhhh... I am sure that I connect correctly to the device. 
I also just tried to reset the device and re-pair it. 

Unfortunately I got the same result.

**Comment by mdima** [comment]
Ok, I solved it. 
I realized that the connection must start from the BT device, not from the smartphone. Before I started to connect from the smartphone to the device, and I got the result above. 
Now I just wait for the connection to be established from the BT device, and the device sends the data I expect. 

Sorry @peterfoot if I made you waste time, I thank you anyway for your support and close the issue. 

Best regards,
Michele
